crates/dbsp/src/circuit/runtime.rs

```diff
@@ -715,18 +657,23 @@ impl RuntimeHandle {
         #[allow(clippy::needless_collect)]
         let results: Vec<ThreadResult<()>> = self.workers.into_iter().map(|h| h.join()).collect();
 
-        // Dropping the background channels here is important because it will drop the
-        // last references the communication channel with the background threads and
-        // so ensures they error out during blocking receive
-        self.runtime.local_store().retain(|kv| {
-            kv.downcast_key_ref::<BackgroundChannel>()
-                .map_or_else(|| true, |_| false)
-        });
-        let _background_results: Vec<ThreadResult<()>> = self
-            .background_workers
-            .into_iter()
-            .map(|h| h.join())
-            .collect();
+        // Wait for the background threads. They will exit automatically without
+        // explicit signaling from us because the worker threads removed all of
+        // their background work.
```

Comment:
I guess this answers my earlier comment

Reply:
It worked for me so far (famous last words).